server/api/migrations/versions/650f0ce2da6f_add_alert_activation.py

```diff
@@ -0,0 +1,119 @@
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""add alert activation
+
+Revision ID: 650f0ce2da6f
+Revises: fcf2ea01f99a
+Create Date: 2024-10-30 16:38:07.592754
+
+"""
+
+import os
+from datetime import datetime
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import mysql
+
+import server.api.crud.alert_activation
+
+# revision identifiers, used by Alembic.
+revision = "650f0ce2da6f"
+down_revision = "fcf2ea01f99a"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "alert_activation",
+        sa.Column("id", sa.Integer(), nullable=False),
+        sa.Column(
+            "activation_time", mysql.DATETIME(timezone=True, fsp=3), nullable=False
+        ),
+        sa.Column(
+            "name", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "project", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column("data", sa.JSON(), nullable=True),
+        sa.Column(
+            "entity_id", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "entity_kind",
+            sa.String(length=255, collation="utf8mb3_bin"),
+            nullable=False,
+        ),
+        sa.Column(
+            "event_kind", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "severity", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column("number_of_events", sa.Integer(), nullable=False),
+        sa.PrimaryKeyConstraint("activation_time", "id", name="_alert_activation_uc"),
+    )
+    op.create_index(
+        "ix_alert_activation_activation_time",
+        "alert_activation",
+        ["activation_time"],
+        unique=False,
+    )
+    op.create_index(
+        "ix_alert_activation_project_name",
+        "alert_activation",
+        ["project", "name"],
+        unique=False,
+    )
+    # ### end Alembic commands ###
+
+    partition_interval = os.getenv("PARTITION_INTERVAL", "YEARWEEK").upper()
+    valid_partition_intervals = ["DAY", "YEARWEEK", "MONTH"]
+
+    # Validate the partition interval
+    if partition_interval not in valid_partition_intervals:
+        raise ValueError(
+            f"Partition interval can only be one of the following: {valid_partition_intervals}"
+        )
+
+    # Current UTC date
+    now_utc = datetime.utcnow()
```

Comment:
This value is going to be used as the `LESS THAN` value in the partition created. Of course, we're not directly using it but rather its translation to day/month/week, but it's still possible that the only partition created at this step will be already expired immediately after - this will lead into SQL errors when trying to insert to the table until a new partition is created.
Since we're using `LESS THAN`, it makes sense to create the 1st partition to be less than now + (partition interval), so the partition will just start at `now`, not end at it.

Another option is to create a "catchall" partition using `LESS THAN MAXVALUE` to handle situations where a partition is not created for some reason. See https://dev.mysql.com/doc/refman/8.0/en/partitioning-range.html for details. I think we should not go for this approach, but it's something to keep in mind.

Reply:
@theSaarco the idea of this PR is to create a migration which creates a table. That's mysql limitation that partitioned table can't exist without at least 1 partition. As a part of next ticket, the logic of creating partitions on start of chief for the next weeks will be implemented, so we will be safe there.